src/common/categorical.h

```diff
@@ -83,7 +83,7 @@ inline void InvalidCategory() {
  * \brief Whether should we use onehot encoding for categorical data.
  */
 XGBOOST_DEVICE inline bool UseOneHot(uint32_t n_cats, uint32_t max_cat_to_onehot, ObjInfo task) {
```

Comment:
Remove task argument. I think with this change task can be removed from a lot of function arguments.

Reply:
Thank you for reminding me of that. All removed.